kim/src/BJ1477.java

```diff
@@ -0,0 +1,63 @@
+import java.io.*;
+import java.util.*;
+
+public class BJ1477 {
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int n = parse(st.nextToken()); // 현재 휴게소의 개수
+        int m = parse(st.nextToken()); // 더 지으려고 하는 휴게소의 개수
+        int l = parse(st.nextToken()); // 고속도로의 길이
+
+        int[] currentArea = new int[n + 2];
+        currentArea[0] = 0;
+        currentArea[n + 1] = l;
+        st = new StringTokenizer(br.readLine());
+        for (int i = 1; i <= n; i++) {
+            currentArea[i] = parse(st.nextToken());
+        }
+        Arrays.sort(currentArea);
+
+        PriorityQueue<RestDist> pq = new PriorityQueue<>();
+        for (int i = 1; i <= n + 1; i++) {
+            pq.offer(new RestDist(currentArea[i] - currentArea[i - 1], 1));
+        }
+
+        while (m-- > 0) {
+            RestDist rd = pq.poll();
+            rd.n++;
```

Comment:
`rd.n++` 도 RestDist 의 책임으로 넘기면 코드가 좀더 깔끔할 것 같습니다. (이 부분은 PS의 영역은 아니긴 합니다만) 이미 다른 부분도 RestDict 의 책임으로 값을 호출하는 것 처럼 보여서요.
```suggestion
            rd.increase();
            
       private static class RestDist implements Comparable<RestDist> {
           
           public void increase() {
              this.n++;
           }
       }
```

Reply:
이런 조언 너무 좋습니다 감사합니다~